#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_TcpClient_IPv6_r122598.dpatch by David Paleino <d.paleino@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch backported from Mono trunk/ r122598
## DP: Fixes IPv6 in TcpClient()

@DPATCH@
diff -urNad mono-2.0.1~/mcs/class/System/System.Net.Sockets/TcpClient.cs mono-2.0.1/mcs/class/System/System.Net.Sockets/TcpClient.cs
--- mono-2.0.1~/mcs/class/System/System.Net.Sockets/TcpClient.cs	2008-09-02 17:45:39.000000000 +0200
+++ mono-2.0.1/mcs/class/System/System.Net.Sockets/TcpClient.cs	2009-01-07 09:17:49.000000000 +0100
@@ -84,7 +84,10 @@
 			}
 			
 			Init (family);
-			client.Bind (new IPEndPoint (IPAddress.Any, 0));
+			IPAddress any = IPAddress.Any;
+			if (family == AddressFamily.InterNetworkV6)
+				any = IPAddress.IPv6Any;
+			client.Bind (new IPEndPoint (any, 0));
 		}
 #endif
 		
